### ISSUE TITLE
Use normal futures instead of preview?

### ISSUE THREAD

**Issue by Hawk777** [sagebind/isahc#125]
So, the [futures crate](https://crates.io/crates/futures) has released version 0.3. Does that mean Isahc should start depending on regular futures instead of futures-preview? The choice between the two is a little bit viral in that it seems I cannot use `AsyncReadExt` functions from `futures` with Isahc’s `Body` class because the latter implements `AsyncRead` from futures-preview instead of futures. Though perhaps I misunderstand the situation.

**Comment by sagebind** [comment]
See #116. Agreed with all these points, though the following should be noted:

- Upgrading from futures-preview to futures 0.3 is a breaking change. Anything using Isahc currently with futures-preview would break by upgrading to futures 0.3 unless they _also_ upgrade.
- Several pieces of the async ecosystem have not moved to futures 0.3 yet, which means that Isahc would not play well with those libraries.

In other words, we are 100% going to do this, but I don't want to do it too quickly, since either way someone's code will be broken. I'd like to release all the pending feature requests under 0.7 before jumping to 0.8.

**Comment by yoshuawuyts** [comment]
@sagebind I'm looking into upgrading Surf in the nearish term to make it work with the new Tide release -- having `isahc` on futures 0.3 would be very helpful for us.

**Comment by sagebind** [comment]
One of the pending features also involves a breaking change, so we'll release all of it together as 0.8. I'll likely have time to release this in a day or so.

**Comment by sagebind** [comment]
Good news, 0.8.0 has been published with the migration to futures 0.3! Enjoy!

**Comment by yoshuawuyts** [comment]
@sagebind thanks so much; we've landed https://github.com/http-rs/surf/pull/111 to upgrade. Very excited!
